### ISSUE TITLE
Error when creating instance of new registered API

### ISSUE THREAD

**Issue by devdattakulkarni** [cloud-ark/kubeplus#771]
Ran into following issue on Minikube.

kubectl create -f wordpress-service-composition.yaml
kubectl create ns wp-stack-ns1
kubectl create -f tenant1.yaml 
kubectl logs kubeplus -c crd-hook

```
..inside queryKubeDiscoveryServiceGet "http://10.96.125.86:90/apis/kubeplus/deploy?platformworkflow=wordpress-service-composition&customresource=wp-service-tenant1&namespace=wp-stack-ns1&overrides={\"namespace\":\"wp-stack-ns1\",\"nodeName\":\"minikube\",\"tenantName\":\"tenant1\"}": dial tcp 10.96.125.86:90: i/o timeout
2021/03/27 10:16:21 http: panic serving 10.0.2.15:35732: runtime error: invalid memory address or nil pointer dereference
goroutine 31892 [running]:
net/http.(*conn).serve.func1(0xc0000ad540)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:1800 +0x139
panic(0x11c0f60, 0x1d96800)
	/usr/local/Cellar/go/1.14.5/libexec/src/runtime/panic.go:975 +0x3e3
main.queryKubeDiscoveryService(0xc0003140f0, 0xe7, 0x0, 0x0, 0x0)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/utils.go:913 +0x2b3
main.QueryDeployEndpoint(0xc000289bbf, 0x1d, 0xc0003db167, 0x12, 0xc0003db188, 0xc, 0xc0000f4050, 0x49, 0x0, 0x0, ...)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/utils.go:815 +0x429
main.handleCustomAPIs(0xc0004e81b0, 0x1dd6148)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:927 +0xea2
main.(*WebhookServer).mutate(0xc0000b0318, 0xc0004e81b0, 0xc0003a192a, 0x6, 0x1)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:155 +0x67c
main.(*WebhookServer).serve(0xc0000b0318, 0x14d51c0, 0xc000136000, 0xc00053bd00)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:1222 +0x891
net/http.HandlerFunc.ServeHTTP(0xc0000496e0, 0x14d51c0, 0xc000136000, 0xc00053bd00)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2041 +0x44
net/http.(*ServeMux).ServeHTTP(0xc0000b3c40, 0x14d51c0, 0xc000136000, 0xc00053bd00)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2416 +0x1a5
net/http.serverHandler.ServeHTTP(0xc000136620, 0x14d51c0, 0xc000136000, 0xc00053bd00)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2836 +0xa3
net/http.(*conn).serve(0xc0000ad540, 0x14d7140, 0xc000398d80)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:1924 +0x86c
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2962 +0x35c
```

**Comment by devdattakulkarni** [comment]
```
=== User ===
Inside handleDelete...
&&&&&
APIv1:platformapi.kubeplus/v1alpha1, Kind=WordpressService, APIv2:{platformapi.kubeplus v1alpha1 wordpressservices}
Group:platformapi.kubeplus, version:v1alpha1
Inside GetPlural...
..Inside getServiceEndpoint...
Host:10.96.125.86, Port:90
After getServiceEndpoint...
Url:http://10.96.125.86:90/apis/kubeplus/getPlural?kind=WordpressService&group=platformapi.kubeplus
2021/03/27 10:26:02 sending request failed: Get "http://10.96.125.86:90/apis/kubeplus/getPlural?kind=WordpressService&group=platformapi.kubeplus": dial tcp 10.96.125.86:90: i/o timeout
..inside queryKubeDiscoveryServiceGet "http://10.96.125.86:90/apis/kubeplus/getPlural?kind=WordpressService&group=platformapi.kubeplus": dial tcp 10.96.125.86:90: i/o timeout
2021/03/27 10:26:02 http: panic serving 10.0.2.15:40160: runtime error: invalid memory address or nil pointer dereference
goroutine 55295 [running]:
net/http.(*conn).serve.func1(0xc0000ad7c0)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:1800 +0x139
panic(0x11c0f60, 0x1d96800)
	/usr/local/Cellar/go/1.14.5/libexec/src/runtime/panic.go:975 +0x3e3
main.queryKubeDiscoveryService(0xc0004b78c0, 0x5f, 0x0, 0x0, 0x0)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/utils.go:913 +0x2b3
main.GetPlural(0xc000046b70, 0x10, 0xc0003509e0, 0x14, 0xc00015f568, 0x2, 0x2)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/utils.go:803 +0x393
main.handleDelete(0xc0002b11a0)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:217 +0x414
main.(*WebhookServer).mutate(0xc0000b0318, 0xc0002b11a0, 0xc000046c5a, 0x6, 0x1)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:121 +0xde4
main.(*WebhookServer).serve(0xc0000b0318, 0x14d51c0, 0xc000136000, 0xc000525400)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:1222 +0x891
net/http.HandlerFunc.ServeHTTP(0xc0000496e0, 0x14d51c0, 0xc000136000, 0xc000525400)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2041 +0x44
net/http.(*ServeMux).ServeHTTP(0xc0000b3c40, 0x14d51c0, 0xc000136000, 0xc000525400)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2416 +0x1a5
net/http.serverHandler.ServeHTTP(0xc000136620, 0x14d51c0, 0xc000136000, 0xc000525400)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2836 +0xa3
net/http.(*conn).serve(0xc0000ad7c0, 0x14d7140, 0xc000465440)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:1924 +0x86c
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2962 +0x35c
```

**Comment by devdattakulkarni** [comment]
```
kubectl logs kubeplus -c helmer
Before registering...

```

**Comment by devdattakulkarni** [comment]
```
2021/03/27 10:47:14 sending request failed: Get "http://10.106.227.214:90/apis/kubeplus/deploy?platformworkflow=wordpress-service-composition&namespace=default&dryrun=true": dial tcp 10.106.227.214:90: i/o timeout
Get "http://10.106.227.214:90/apis/kubeplus/deploy?platformworkflow=wordpress-service-composition&namespace=default&dryrun=true": dial tcp 10.106.227.214:90: i/o timeout
2021/03/27 10:47:14 http: panic serving 10.0.2.15:41532: runtime error: invalid memory address or nil pointer dereference
goroutine 4753 [running]:
net/http.(*conn).serve.func1(0xc000269360)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:1800 +0x139
panic(0x11c0f60, 0x1d96800)
	/usr/local/Cellar/go/1.14.5/libexec/src/runtime/panic.go:975 +0x3e3
main.queryKubeDiscoveryService(0xc0004be680, 0x7a, 0x0, 0x0, 0x0)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/utils.go:913 +0x2b3
main.DryRunChart(0xc00051b3bf, 0x1d, 0x131b133, 0x7, 0xc0004b2088, 0xc00038fe01, 0x8)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/utils.go:827 +0x393
main.getPaCAnnotation(0xc0002ae1e0, 0x132951d)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:800 +0x65a
main.(*WebhookServer).mutate(0xc00000e2b0, 0xc0002ae1e0, 0xc0002a6780, 0x6, 0x1)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:146 +0xd5f
main.(*WebhookServer).serve(0xc00000e2b0, 0x14d51c0, 0xc0004a20e0, 0xc00010f800)
	/Users/devdatta/go/src/github.com/cloud-ark/kubeplus/mutating-webhook/webhook.go:1222 +0x891
net/http.HandlerFunc.ServeHTTP(0xc00008da30, 0x14d51c0, 0xc0004a20e0, 0xc00010f800)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2041 +0x44
net/http.(*ServeMux).ServeHTTP(0xc0000b2e80, 0x14d51c0, 0xc0004a20e0, 0xc00010f800)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2416 +0x1a5
net/http.serverHandler.ServeHTTP(0xc0004a2460, 0x14d51c0, 0xc0004a20e0, 0xc00010f800)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2836 +0xa3
net/http.(*conn).serve(0xc000269360, 0x14d7140, 0xc000541840)
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:1924 +0x86c
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.14.5/libexec/src/net/http/server.go:2962 +0x35c
/mutate

```

**Comment by devdattakulkarni** [comment]
Fixed in #825
